tests/pipelines/test_pipelines_text_to_audio.py

```diff
@@ -0,0 +1,242 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+import numpy as np
+
+from transformers import (
+    MODEL_FOR_TEXT_TO_WAVEFORM_MAPPING,
+    AutoProcessor,
+    TextToAudioPipeline,
+    pipeline,
+)
+from transformers.testing_utils import (
+    is_pipeline_test,
+    require_torch,
+    require_torch_gpu,
+    require_torch_or_tf,
+    slow,
+)
+
+from .test_pipelines_common import ANY
+
+
+@is_pipeline_test
+@require_torch_or_tf
+class TextToAudioPipelineTests(unittest.TestCase):
+    model_mapping = MODEL_FOR_TEXT_TO_WAVEFORM_MAPPING
+    # for now only text_to_waveform and not text_to_spectrogram
+
+    @slow
+    @require_torch
+    def test_small_model_pt(self):
+        speech_generator = pipeline(task="text-to-audio", model="facebook/musicgen-small", framework="pt")
+
+        forward_params = {
+            "do_sample": False,
+            "max_new_tokens": 250,
+        }
+
+        outputs = speech_generator("This is a test", forward_params=forward_params)
+
+        # musicgen sampling_rate is not straightforward to get
+        self.assertIsNone(outputs["sampling_rate"])
+
+        audio = outputs["audio"]
+
+        self.assertEqual(
+            ANY(np.ndarray),
+            audio,
+        )
+
+        # test two examples side-by-side
+        outputs = speech_generator(["This is a test", "This is a second test"], forward_params=forward_params)
+
+        audio = [output["audio"] for output in outputs]
+
+        self.assertEqual(
+            [
+                ANY(np.ndarray),
+                ANY(np.ndarray),
+            ],
+            audio,
+        )
+
+        # test batching
+        outputs = speech_generator(
+            ["This is a test", "This is a second test"], forward_params=forward_params, batch_size=2
+        )
+
+        self.assertEqual(
+            ANY(np.ndarray),
+            outputs[0]["audio"],
+        )
+
+    @slow
+    @require_torch
+    def test_large_model_pt(self):
+        speech_generator = pipeline(task="text-to-audio", model="suno/bark-small", framework="pt")
+
+        # test text-to-speech
+
+        forward_params = {
+            # Using `do_sample=False` to force deterministic output
+            "do_sample": False,
+            "semantic_max_new_tokens": 100,
+        }
+
+        # atm, must do to stay coherent with BarkProcessor
+        preprocess_params = {
+            "max_length": 256,
+            "add_special_tokens": False,
+            "return_attention_mask": True,
+            "return_token_type_ids": False,
+            "padding": "max_length",
+        }
```

Comment:
At the moment, **one of the limitation of the TTS pipeline** is that the user must do some of the operations that were initially done in the `Processor`.

For example, `BarkProcessor` passes this `preprocess_params` [under-the-hood](https://github.com/huggingface/transformers/blob/972fdcc77878cf7afcc8aef8979d6b4241005bb6/src/transformers/models/bark/processing_bark.py#L272-L281).

I'm not sure yet how to best correct this behaviour, which adds unnecessary and unclear difficulty to the user. Some models can't be used without those parameters.

Reply:
I guess one way to solve it for Bark would be to create a BarkTokenizer which call BertTokenizer under-the-hood and set these parameters.